src/K8s/WsSwoole/ClientFactory.php

```diff
@@ -20,16 +20,21 @@ class ClientFactory
 {
     public function makeCoroutineClient(RequestInterface $request, array $options = []): Client
     {
+        $ssl = $request->getUri()->getScheme() === 'wss';
+
         $client = new Client(
             $request->getUri()->getHost(),
-            $request->getUri()->getPort(),
-            $request->getUri()->getScheme() === 'wss'
+            $request->getUri()->getPort() ?? ($ssl ? 443 : 80),
+            $ssl
         );
 
         if (!empty($options)) {
             $client->set($options);
         }
-        $client->setHeaders($request->getHeaders());
+
+        $client->setHeaders(array_map(function ($value) {
+            return $value[0];
+        }, $request->getHeaders()));
```

Comment:
What is the reason for this addition? The other change seems good. But wont this pass just an array of header values missing the key for the header name?

Reply:
The key-value pairs of the parameters for `Swoole\Coroutine\Http\Client::setHeaders` must be strings , cannot be arrays.

Reply:
I think instead of `return $value[0]` it should probably be something like `return implode(",", value);`, as multiple values for a a header is allowed.